src/DataProtection/DataProtection/src/Repositories/RegistryXmlRepository.cs

```diff
@@ -155,4 +155,62 @@ private void StoreElementCore(XElement element, string valueName)
         // but the window for that should be small enough that we shouldn't have to worry about it.
         RegistryKey.SetValue(valueName, element.ToString(), RegistryValueKind.String);
     }
+
+    /// <inheritdoc/>
+    public virtual bool RemoveElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(chooseElements);
+
+        var deletableElements = new List<DeletableElement>();
+
+        foreach (var valueName in RegistryKey.GetValueNames())
+        {
+            var element = ReadElementFromRegKey(RegistryKey, valueName);
+            if (element is not null)
+            {
+                deletableElements.Add(new DeletableElement(valueName, element));
+            }
+        }
+
+        chooseElements(deletableElements);
+
+        var elementsToDelete = deletableElements
+            .Where(e => e.DeletionOrder.HasValue)
+            .OrderBy(e => e.DeletionOrder.GetValueOrDefault());
+
+        foreach (var deletableElement in elementsToDelete)
+        {
+            var valueName = deletableElement.ValueName;
+            _logger.RemovingDataFromRegistryKeyValue(RegistryKey, valueName);
+            try
+            {
+                RegistryKey.DeleteValue(valueName, throwOnMissingValue: false);
+            }
+            catch (Exception ex)
+            {
+                _logger.FailedToRemoveDataFromRegistryKeyValue(RegistryKey, valueName, ex);
+                return false;
```

Comment:
```suggestion
                // Stop processing deletions to avoid deleting a revocation entry for a key that we failed to delete.
                return false;
```